File: app/models/dive_group_participant.rb
class DiveGroupParticipant < Synced
  attr_accessible :dive_group_id, :dive_role_id, :diver_id, :created_by_app_key_id, :last_updated_by_app_key_id
  
  belongs_to :dive_group
  belongs_to :diver
  belongs_to :dive_role
  
  validates :dive_group_id, :dive_role_id, :diver_id, :presence => true
  
  def self.findCreatedDiff(app_key_id, sync_date)
	if not sync_date.nil?
		conditions = [" (dive_group_participants.created_by_app_key_id IS NULL OR dive_group_participants.created_by_app_key_id  <> ?)
						AND (dive_group_participants.created_at > ?)", 
					  app_key_id, 
					  sync_date]
	else
		conditions = [" (dive_group_participants.created_by_app_key_id IS NULL OR dive_group_participants.created_by_app_key_id  <> ?) ", app_key_id]
	end
	return self.find(:all, :conditions => conditions)
  end
  
  def self.findUpdatedDiff(app_key_id, sync_date)
	return self.find(
					:all, 
					:conditions => [
									" (dive_group_participants.last_updated_by_app_key_id IS NULL OR dive_group_participants.last_updated_by_app_key_id <> ?)
									  AND (dive_group_participants.updated_at > ?) 
									  AND (dive_group_participants.created_at <> dive_group_participants.updated_at)", 
									  app_key_id,												  
									  sync_date
									])
  end
  
end

File: app/assets/javascripts/app/models/dive-group-participant.js
#= require ../models.js
#= require ../lib/resource/synced.js

JustDive.Models.DiveGroupParticipant = JustDive.Resource.Synced.extend({
	resourceUrl: 			'/dive_group_participants',
	resourceName:       	'dive_group_participant',
	resourceProperties: [
							'id', 
							'dive_group_id',
							'dive_role_id',
							'diver_id',
							'created_at',
							'updated_at'
						],
						
	title: 				Ember.computed(function() {
							return this.get('id');
						}).property('id'),
	
	diver: 				Ember.computed(function() {
							var diverId = this.get('diver_id') || undefined;
							if (diverId === undefined) {
								return false;
							} else {
								return JustDive.restControllers.divers.findObject(diverId);
							}
						}).property('diver_id').cacheable(),
	
	diveRole: 			Ember.computed(function() {
							var diveRoleId = this.get('dive_role_id') || undefined;
							if (diveRoleId === undefined) {
								return false;
							} else {
								return JustDive.restControllers.dive_roles.findObject(diveRoleId);
							}
						}).property('dive_role_id').cacheable(),
	
	diveGroup: 			Ember.computed(function() {
							var diveGroupId = this.get('dive_group_id') || undefined;
							if (diveGroupId === undefined) {
								return false;
							} else {
								return JustDive.restControllers.dive_groups.findObject(diveGroupId);
							}
						}).property('dive_group_id').cacheable(),
});
